src/Units/UnitModel.java

```diff
@@ -197,6 +197,26 @@ public boolean hasDirectFireWeapon()
     return hasDirect;
   }
 
+  /**
+   * @return True if this UnitModel has at least one weapon that cannot fire after moving.
+   */
+  public boolean hasImmobileWeapon()
```

Comment:
This name feels pretty unintuitive. I don't anticipate asking the question "Does my unit have a weapon that it can't use after moving?" - a much more likely question is "Does this unit have a siege/ranged weapon?"

Is there a reason not to rename this to `hasRangedWeapon`? I know those aren't exactly the same question (they would have different answers for BShips that can move and shoot), but I'm not sure the current name is the more useful/relevant one.

Reply:
Yes, there's a reason.
Immobile weapons, by their nature, cannot reposition to attack something that's not immediately in range.
I want to make sure any "siege"/immobile weapons get a chance to attack first since they are liable to be powerful and will free up attacking space for mobile attackers... and because more attacks is more attacks, period.

Immobile weapons are un-complicated to calculate, and it's important that they attack if possible. Ranged weapons (e.g. DoR battleships) can be *more* complicated to calculate, since they have so many valid places to attack from (and Wally's code fully accounts for that, when trying for N-hit KOs).